.ci.yaml

```diff
@@ -3706,6 +3706,7 @@ targets:
       tags: >
         ["devicelab", "ios", "mac"]
       task_name: flutter_gallery__transition_perf_e2e_ios
+      artifact: gallery__transition_perf_e2e_ios
```

Comment:
These artifacts can't be shared, `flutter_gallery__transition_perf_e2e_impeller_ios` it built with impeller and `flutter_gallery__transition_perf_e2e_ios` is not.

It seems like we should have some more generic way to do this, like hashing the example/integration_test app directory and the build flags?  Also, some tests build the app multiple times with different flags (debug, profile, release).

Reply:
Right, different tasks with different building args are not sharing these artifact.
I am planning to drop this `artifact` property for non-sharing ones. If two tasks do share the same artifact, we will specify the artifact explicitly.

Reply:
Latest change: removed the artifact property here. In general, if `artifact` is not specified, `task name` will be used as `artifact` for the task. On the other side, if specified, the task will use it, and other task will share it if same `artifact` is defined.

This way, we have an assumption that each task builds a single artifact. This way, tasks that follow build+test+build+test etc. will not be supported for the later artifact sharing. As talked in go/devicelab-build-test, the later build should be covered in the `test` step.

Please let me know if this makes sense to you.